packages/api/src/controllers/helpers.ts

```diff
@@ -699,6 +699,33 @@ export const triggerCatalystPullStart =
         throw new Error(`failed to trigger catalyst pull`);
       };
 
+export const getCatalystStreamMetadata =
+  process.env.NODE_ENV === "test"
+    ? async () => {} // noop in case of tests
+    : async (url: string) => {
+        const deadline = Date.now() + 2 * PULL_START_TIMEOUT;
+        while (Date.now() < deadline) {
+          const res = await fetchWithTimeoutAndRedirects(url, {
+            method: "GET",
+            timeout: PULL_START_TIMEOUT,
+            maxRedirects: 10,
+          });
+          const body = await res.text();
+          if (res.ok && body.includes(`"meta":`)) {
```

Comment:
Does the stream always include `meta: ` or can there be a case in which the playback works but the `meta: ` is not returned?

I'm trying to think if there is any possibility of breaking of what we currently have.

Reply:
That came from Jaron, so hopefully we're good:
> If there's a meta member, the stream is active

Reply:
Jaron confirmed that `meta` is only missing in these cases:
> Yeah. It's only missing while the stream is offline or in the middle of booting/shutting down